react/features/authentication/middleware.ts

```diff
@@ -100,26 +108,51 @@ MiddlewareRegistry.register(store => next => action => {
         if (_isWaitingForOwner(store)) {
             store.dispatch(stopWaitForOwner());
         }
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONFERENCE_LEFT:
         store.dispatch(stopWaitForOwner());
         break;
 
     case CONNECTION_ESTABLISHED:
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONNECTION_FAILED: {
         const { error } = action;
+        const state = store.getState();
+        const { jwt } = state['features/base/jwt'];
 
         if (error
                 && error.name === JitsiConnectionErrors.PASSWORD_REQUIRED
-                && typeof error.recoverable === 'undefined') {
+                && typeof error.recoverable === 'undefined'
+                && !jwt) {
```

Comment:
Sorry for the newbie question but why do we need to check for jwt here?

Reply:
As there are different cases where you will receive PASSWORD_REQUIRED if you have setup deployment with username and password and you try to join or supply wrong username or password. Or if you configure the deployment to require jwt tokens and do not allow guests without tokens and you try to connect without a token.